src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1242,6 +1237,8 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
                         }
                     }
 
+                    explicitInterfaceImplementations = explicitInterfaceImplementationsBuilder.ToImmutableAndFree();
```

Comment:
>explicitInterfaceImplementations [](start = 20, length = 32)

Would it be easier to remove loops entirely from the method, to avoid the complexity of handling both in the same loop:
```
explicitInterfaceImplementations = explicitlyOverriddenMethods.WhereAsArray(m => m.ContainingType.IsInterface);
MethodSymbol uniqueClassOverride = getUniqueClassOverride(explicitlyOverriddenMethods);

static MethodSymbol getUniqueClassOverride(ImmutableArray<MethodSymbol> explicitlyOverriddenMethods) { ... }
```

Reply:
> We're skipping any trailing explicit interface implementations. Is that an issue? Consider adding a test.
> 
> Refers to: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:1233 in [fd91c43](https://github.com/dotnet/roslyn/commit/fd91c43b5c62c8c0b7b6f4402ccce0fa417d6f93). [](commit_id = [fd91c43](https://github.com/dotnet/roslyn/commit/fd91c43b5c62c8c0b7b6f4402ccce0fa417d6f93), deletion_comment = False)

Thanks for catching this. I did change semantics by accident. 😉 I am thinking perhaps we should try @davidwrighton's suggestion for now to just apply `[MethodImpl(MethodImplOptions.NoOptimization)]` to this method as well as perhaps a `WorkItem` link to track removing it once the related JIT bug is fixed.